spec/jobs/socure_shadow_mode_proofing_job_spec.rb

```diff
@@ -0,0 +1,384 @@
+                  success: true,
+                  timed_out: false,
+                  transaction_id: '',
+                  vendor_name: 'ResidentialAddressNotRequired',
+                  vendor_workflow: nil,
+                  verified_attributes: nil,
+                },
+                resolution: {
+                  attributes_requiring_additional_verification: [],
+                  can_pass_with_additional_verification: false,
+                  errors: {},
+                  exception: nil,
+                  reference: 'aaa-bbb-ccc',
+                  success: true,
+                  timed_out: false,
+                  transaction_id: 'resolution-mock-transaction-id-123',
+                  vendor_name: 'ResolutionMock',
+                  vendor_workflow: nil,
+                  verified_attributes: nil,
+                },
+                state_id: {
+                  errors: {},
+                  exception: nil,
+                  mva_exception: nil,
+                  requested_attributes: {},
+                  success: true,
+                  timed_out: false,
+                  transaction_id: 'state-id-mock-transaction-id-456',
+                  vendor_name: 'StateIdMock',
+                  verified_attributes: [],
+                },
+                threatmetrix: {
+                  client: nil,
+                  errors: {},
+                  exception: nil,
+                  review_status: 'pass',
+                  success: true,
+                  timed_out: false,
+                  transaction_id: 'ddp-mock-transaction-id-123',
+                },
+              },
+            },
+            exception: nil,
+            ssn_is_unique: true,
+            threatmetrix_review_status: 'pass',
+            timed_out: false,
+          },
+          socure_result: {
+            attributes_requiring_additional_verification: [],
+            can_pass_with_additional_verification: false,
+            errors: { reason_codes: ['I905', 'I914', 'I919'] },
+            exception: nil,
+            reference: '',
+            success: true,
+            timed_out: false,
+            transaction_id: 'a1234b56-e789-0123-4fga-56b7c890d123',
+            vendor_name: 'socure_kyc',
+            vendor_workflow: nil,
+            verified_attributes: %i[address first_name last_name phone ssn dob].to_set,
+          },
+        )
+      end
+
+      context 'when socure proofer raises an error' do
+        before do
+          allow(job.proofer).to receive(:proof).and_raise
+        end
+
+        it 'does not squash the error' do
+          # The Proofer converts errors raised during requests into results
+          # with the `exception` property set. Any other error
+          expect { perform }.to raise_error
+        end
+      end
+    end
+
+    context 'when document_capture_session_result_id is not valid' do
+      let(:document_capture_session_result_id) { 'some-id-that-is-not-valid' }
```

Comment:
I think I'm just being dense, but what does "not valid" mean here?

I _think_ you mean something like "When we cannot find the result" or "When the result itself is invalid" (?), but my brain is interpreting this as more like "When the result_id parameter is formatted incorrectly."

Reply:
Yes, that's a good callout. In this case it just means it's not present in redis. I'll update